### PR TITLE
[8.0 incubator] medical patient occupation

### PR DESCRIPTION
**Pull request by lasley**
Migrate medical_emr/medical_occupation to medical_patient_occupation and update API

### CODE REVIEW

**Comment by lasley** [comment]
Hey all - I keep getting hung on the RunBot checks. Is there some sort of local version of this that I can start using? I tried finding it in the maintainer-tools repo, but I don't see anything that stands out.

File: medical_patient_occupation/models/medical_patient.py [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+###############################################################################
+
+from openerp import fields, models
+
+
+class MedicalPatient(models.Model):
+    _inherit = 'medical.patient'
+    occupation_id = fields.Many2one('medical.patient.occupation')
```
Comment (JonathanNEMRY) [inline medical_patient_occupation/models/medical_patient.py]: add a string attribute "Occupation"

File: medical_patient_occupation/models/medical_patient_occupation.py [hunk 2]
```diff
@@ -21,17 +21,15 @@
...
 
-from openerp.osv import fields, orm
+from openerp import fields, models
 
 
-class MedicalOccupation(orm.Model):
-    _name = 'medical.occupation'
+class MedicalPatientOccupation(models.Model):
+    _name = 'medical.patient.occupation'
+
```
Comment (JonathanNEMRY) [inline medical_patient_occupation/models/medical_patient_occupation.py]: add a _description = "Medical Occupation"

File: medical_patient_occupation/models/medical_patient_occupation.py [hunk 3]
```diff
@@ -21,17 +21,15 @@
...
+    code = fields.Char(size=256, )
```
Comment (JonathanNEMRY) [inline medical_patient_occupation/models/medical_patient_occupation.py]: remove size attribute

File: medical_patient_occupation/models/medical_patient_occupation.py [hunk 4]
```diff
@@ -21,17 +21,15 @@
...
+    name = fields.Char(size=256, required=True, translate=True, )
 
-    _columns = {
-        'code': fields.char(size=256, string='Code'),
-        'name': fields.char(size=256, string='Name', required=True,
-                            translate=True),
-    }
     _sql_constraints = [
         ('name_uniq', 'UNIQUE(name)', 'Name must be unique!'),
```
Comment (JonathanNEMRY) [inline medical_patient_occupation/models/medical_patient_occupation.py:33]: I see you add a sql_constraint on code for this one: https://github.com/OCA/vertical-medical/pull/58/files Is there a reason to not doing the same here?

File: medical_patient_occupation/views/medical_patient_occupation_view.xml [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+		</record>
+
+		<record id="medical_patient_occupation_view_form" model="ir.ui.view">
+			<field name="name">medical.patient.occupation.form</field>
+			<field name="model">medical.patient.occupation</field>
+			<field name="priority" eval="8" />
+			<field name="arch" type="xml">
+				<form string="Patient Occupation">
+					<group>
+						<field name="code" />
```
Comment (JonathanNEMRY) [inline medical_patient_occupation/views/medical_patient_occupation_view.xml]: Better presentation with div title just like Odoo views

File: medical_patient_occupation/views/medical_patient_occupation_view.xml [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+			<field name="name">medical.patient.occupation.search</field>
+			<field name="model">medical.patient.occupation</field>
+			<field name="priority" eval="8" />
+			<field name="arch" type="xml">
+				<search string="Patient Occupation">
+					<field name="code" />
+					<field name="name" />
+					<newline />
+					<group expand="0" string="Group By...">
+						<filter string="Code" domain="[]" context="{'group_by':'code'}" />
```
Comment (JonathanNEMRY) [inline medical_patient_occupation/views/medical_patient_occupation_view.xml]: no need of group search with these kind of fields

File: medical_patient_occupation/views/medical_patient_occupation_view.xml [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+			<field name="type">ir.actions.act_window</field>
+			<field name="res_model">medical.patient.occupation</field>
+			<field name="view_type">form</field>
+			<field name="view_mode">tree,form</field>
+			<field name="search_view_id" ref="medical_patient_occupation_view_search" />
+			<field name="domain">[]</field>
+			<field name="context">{}</field>
+		</record>
+
+		<record model="ir.actions.act_window.view" id="medical_patient_occupation_action_form">
```
Comment (JonathanNEMRY) [inline medical_patient_occupation/views/medical_patient_occupation_view.xml]: Only one ir.actions.act_window per view with view_id=tree and search_view_id=search. So action record for tree and form can be removed

File: medical_patient_occupation/models/medical_patient_occupation.py (hunk 4 above)
Comment (lasley) [inline medical_patient_occupation/models/medical_patient_occupation.py:33]: Yup - from what I understand, there aren't really overlapping Ethnicity codes; but Occupations sometimes don't have codes, or overlap depending on how they're divided. Now that I'm looking though, I also meant to add a `required=True` to ethnicity codes.

File: medical_patient_occupation/views/medical_patient_occupation_view.xml (hunk 7 above)
Comment (lasley) [inline medical_patient_occupation/views/medical_patient_occupation_view.xml]: Just out of curiosity, was this an API change or something? Seems this is done in a lot of the vertical-medical code.

Comment (JonathanNEMRY) [inline medical_patient_occupation/views/medical_patient_occupation_view.xml]: Don't think it is an API change. I don't know why it was implemented like that. Only one required action with a pointer to the tree view and the search view. The form view is implicit.

**Pull request closed without merging**